.github/workflows/package-release.yml

```diff
@@ -0,0 +1,113 @@
+name: "Package Release"
+run-name: "Package Release ${{ github.ref_name }}"
+
+on:
+  push:
+    tags:
+      - "*"
+
+jobs:
+  build-pecl:
+    name: "Create PECL package"
+    runs-on: "ubuntu-latest"
+
+    steps:
+      - name: "Checkout"
+        uses: "actions/checkout@v4"
+        with:
+          # Manually specify a ref. When actions/checkout is run for a tag without a ref, it looks up the underlying
+          # commit and specifically fetches this to the refs/tags/<tag> ref, which denies us access to the tag message
+          ref: ${{ github.ref }}
+          fetch-depth: 2
+          submodules: true
+
+      - name: "Build Driver"
+        uses: ./.github/workflows/linux/build
+        with:
+          version: "8.3"
+
+      - name: "Write changelog file for packaging"
+        run: git tag -l ${{ github.ref_name }} --format='%(contents)' > changelog
+
+      - name: "Build package.xml"
+        run: "make package.xml"
+
+      - name: "Build release archive"
+        run: "make package"
+
+      # PECL always uses the version for the package name.
+      # Read it from the version file and store in env to use when uploading artifacts
+      - name: "Read current package version"
+        run: echo "PACKAGE_VERSION=$(./bin/update-release-version.php version)" >> "$GITHUB_ENV"
+
+      # Currently fails with: Cannot install, php_dir for channel "pecl.php.net" is not writeable by the current user
+      # - name: "Install release archive to verify correctness"
+      #  run: pecl install mongodb-${{ env.PACKAGE_VERSION }}.tgz
+
+      - name: "Upload artifact"
+        uses: actions/upload-artifact@v4
+        with:
+          name: mongodb-${{ env.PACKAGE_VERSION }}.tgz
+          path: mongodb-${{ env.PACKAGE_VERSION }}.tgz
+          retention-days: 3
+
+      - name: "Upload release artifact"
+        run: gh release upload ${{ github.ref_name }} mongodb-${{ env.PACKAGE_VERSION }}.tgz
+        # Until we've changed the process to always have a release draft, ignore errors during upload
+        continue-on-error: true
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  build-windows:
+    name: "Create Windows package"
+    runs-on: windows-2022
+    defaults:
+      run:
+        shell: cmd
+
+    strategy:
+      fail-fast: false
```

Comment:
I used `fail-fast: false` to ensure that if any build errored, other builds would be unaffected. For example, if compilation for PHP 7.4 failed, I wouldn't want to abort all previous builds but rather complete them and only have to revisit the failing one.

I'll note that this shouldn't ever happen as any build process failures would already be detected by the pull request that introduces the breakage; since the tag we're building will only contain the version update commit that didn't run through CI, I'm expecting no breakage here.

Reply:
Oh, I misread this as `fail-fast: true`. I also didn't realize that was the [default behavior](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast).

_Not_ failing fast definitely makes sense. Feel free to resolve after reading.